client/src/components/Calendar/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react';
+import {Icon, TableHeaderCell, Table, Header, Segment } from 'semantic-ui-react';
+import styled from 'styled-components';
+import Moment from 'moment';
+import {extendMoment} from 'moment-range';
+const moment = extendMoment(Moment);
+
+const HeaderWrapper = styled.div`
+    height: 80px;
+    width: 95%;
+    margin: 3%;
+    overflow: hidden;
+`;
+
+const HeaderArrow = {
+    color: "#FA8072", 
+    paddingTop: "5%"
+};
+
+const Title = styled.span`
+    position: absolute;
+    font-family: Helvetica;
+    font-size: 3em;
+    color: #FA8072;
+    text-align: center;
+    margin-left: 20%;
+    margin-top: 5%;
+    font-weight: bold;
+    text-transform: uppercase;
+`;
+
+class Calendar extends Component {
+    constructor(props){
+        super(props);
+        this.state={
+            month: moment().month(),
+            year: moment().year(),        
+        }
+    }
+
+    previousMonth = ()=>{
+        if(this.state.month === 0){
+            this.setState({
+                month: 11,
+                year: this.state.year-1
+            });
+        } else {
+            this.setState({
+                month: this.state.month - 1,
+                year: this.state.year
+            });
+        }
+        console.log('this.state.month: ' + this.state.month);
+    }
+    nextMonth = ()=>{
+        let month, year;
+        if (this.state.month === 11){
+            this.setState({
+                month: 0,
+                year: this.state.year + 1
+            });
+        }
+        else{
+            this.setState({
+                month: this.state.month + 1,
+                year: this.state.year
+            });
+        }
+    }
+
+    render(){
+        let daysOfWeek = ["Sun", "Mon", "Tues", "Wed", "Thurs", "Fri", "Sat"],
+            year = this.state.year,
+            month = this.state.month,
+            startDate = moment([year, month]),
+            firstOfMonth = moment(startDate).clone().startOf('month'),
+            firstOfMonthDate = firstOfMonth.date(),
+            firstOfMonthDay = firstOfMonth.day(),
+            lastOfMonth = moment(startDate).clone().endOf('month'),
+            lastOfMonthDate = lastOfMonth.date(),
+            lastOfMonthDay = lastOfMonth.day(),
+            daysInMonth = moment([year, month]).daysInMonth(),
+            weeksInMonth = Math.floor((firstOfMonthDay + daysInMonth)/7),
+            weeks = [];
+            
+        {   let week = [],
+                daysToAdd,
```

Comment:
A lot of the date logic could be also pulled somewhere else, like a `generateMonth` function that returns the full data structure, in a dateUtilities file or utilities file.

Also - if you put `let week = []` at the beginning of the loop at line 92, you wouldn't have to reinitialize it on line 140.

Reply:
I moved let week [] like you suggested and created a generateMonth function. The function is still in the same file. I'll meet with you or @blaketarter today about how you'd set up the file structure. Thanks!